k2/python/k2/fsa.py

```diff
@@ -1394,7 +1394,7 @@ def set_scores_stochastic_(self, scores) -> None:
 
         # Note we use `to` here since `scores` and `self.scores` may not
         # be on the same device.
-        self.scores = ragged_scores.values.to(self.scores.device)
+        self.scores = ragged_scores.values().to(self.scores.device)
```

Comment:
`ragged_scores` here is an instance of `k2.ragged.RaggedFloat`, which
has a property `values`.

`k2.ragged.RaggedFloat` is a wrapper around `k2.RaggedFloat` and supports autograd.